hoot-core/src/main/cpp/hoot/core/algorithms/AddressParser.cpp

```diff
@@ -0,0 +1,549 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "AddressParser.h"
+
+// hoot
+#include <hoot/core/util/Log.h>
+#include <hoot/core/util/ConfigOptions.h>
+#include <hoot/core/util/LibPostalInit.h>
+#include <hoot/core/algorithms/ExactStringDistance.h>
+#include <hoot/core/util/FileUtils.h>
+#include <hoot/core/OsmMap.h>
+#include <hoot/core/conflate/Address.h>
+
+// libpostal
+#include <libpostal/libpostal.h>
+
+namespace hoot
+{
+
+QMultiMap<QString, QString> AddressParser::_addressTypeToTagKeys;
+
```

Comment:
There was a lot of generic address parsing logic in the poi/poly address score extractor, so moved it here in case we need to use it with other types of conflation.